experimental/level_zero/status_util.h

```diff
@@ -48,6 +48,14 @@ iree_status_t iree_hal_level_zero_result_to_status(
     iree_hal_level_zero_dynamic_symbols_t* syms, ze_result_t result,
     const char* file, uint32_t line);
 
+#define IREE_LEVEL_ZERO_TRY(...)      \
+  do {                                \
+    status = __VA_ARGS__;             \
+    if (!iree_status_is_ok(status)) { \
+      goto cleanup;                   \
```

Comment:
Would it be possible to avoid using `goto` and issue a call to the clean up function directly?

Reply:
I can change it, but in general for error handling in C `goto` is OK. In the function it is used it turned out to have just the one resource, but in functions where multiple resources are managed the `goto` approach is more sane.
I urge you to do a text search for `goto` in the repo and see how many libraries use it for error handling.
I think in C you don't have a better construct for releasing resources after an error. If it were C++, things could be cleaner with RAII.